File: app/src/main/java/com/example/anchornotes/data/db/TagEntity.java
package com.example.anchornotes.data.db;

import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "tags", indices = {@Index(value = {"name"}, unique = true)})
public class TagEntity {
    @PrimaryKey(autoGenerate = true) public long id;
    public String name;

    public TagEntity() {}
    public TagEntity(String name) { this.name = name; }
}

File: app/src/main/java/com/example/anchornotes/data/db/TagEntity.kt
package com.example.anchornotes.data.db

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Tag entity for organizing notes.
 */
@Entity(tableName = "tags")
data class TagEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String
)

